examples/shell-environments/README.md

```diff
@@ -0,0 +1,52 @@
+# shell-environment
+
+> **Warning**
+> If you copy the flake.nix remember to `git add [-N|--intent-to-add] flake.nix`, otherwise it won't work
+
+This example shows how to create a shell environment which
+includes a diverse set of tools:
+
+```sh
+terraform
+wget
+bat
+git
```

Comment:
I don't think we should include `git`, but rather rely on the user's git.
They need to have their own git in their user profile in order to clone new projects, so it's not necessary to provide one here, and having potentially multiple versions of git operating on the same repos is a *bit* asking for trouble, because you're essentially doing a whole bunch of upgrades and downgrades all the time, if you want to look at it that way. Git is of course managed very well, so you might get away with it, but git in a shell could prove problematic later.

How about `nixpkgs-fmt` instead?